lib/load_brocfile.js

```diff
@@ -4,23 +4,76 @@ const path = require('path');
 const findup = require('findup-sync');
 const esmRequire = require('esm')(module);
 
+/**
+ * Require a brocfile via either ESM or TypeScript
+ *
+ * @param {String} brocfilePath The path to the brocfile
+ * @returns {*}
+ */
+function requireBrocfile(brocfilePath) {
+  let brocfile;
+
+  if (brocfilePath.match(/\.ts$/)) {
+    const configPath = findup('tsconfig.json', {
+      nocase: true,
+    });
+
+    let options;
+
+    if (configPath) {
+      options = { project: configPath };
+    } else {
+      options = {
+        // Define default compilerOptions
+        compilerOptions: {
+          declaration: true,
+          strict: true,
+          moduleResolution: 'node',
+          module: 'commonjs',
+          target: 'es5',
+          baseUrl: '.',
+        },
+      };
+    }
+
+    // Ensure any previous .tsx? extensions are removed
+    delete require.extensions['.ts']; // eslint-disable-line node/no-deprecated-api
```

Comment:
If someone has already set up `ts-node` (or otherwise decided to handle `.ts` files), shouldn't we defer to that instead of blowing it away?

Reply:
yeah I realise this was just for tests, so I will add this to the tests instead